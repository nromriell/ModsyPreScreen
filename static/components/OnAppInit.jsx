import React, {useEffect, Fragment} from 'react'
import {connect} from 'react-redux'
import {action_get_remote_products} from "../redux/actions";

/**
 * @author
 * Nathan Romriell - 10/16/2019
 *
 * @info
 * Functional Component - OnAppInit
 *
 * @description
 * Empty class using useEffect hook to dispatch actions on application load
 *
 * @redux-props
 * loadProductData($page) - initial call to remote server to download product details
 *
 * @Hooks
 * useEffect              - Put redux application load calls in this function
 *
 */
const OnAppInit = (props) => {
    useEffect(() => {
       props.loadProductData(1);
    });
    return <Fragment/>
};

const mapDispatchToProps = (dispatch) =>
{
  return {
      loadProductData: (page) => dispatch(action_get_remote_products(page))
  }
};

export default connect(null, mapDispatchToProps)(OnAppInit)