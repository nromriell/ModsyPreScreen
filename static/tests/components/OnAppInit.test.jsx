import React from 'react'
import renderer from 'react-test-renderer'
import {Provider} from 'react-redux'
import OnAppInit from '../../components/OnAppInit'
import configureStore from 'redux-mock-store'

const mockStore = configureStore([]);

describe('OnAppInit Tests', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            data: {loading: false, products:[]},
            scrollTarget: 500
        });
        component = renderer.create(
            <Provider store={store}>
                <OnAppInit/>
            </Provider>
        )
    });

    it('Renders Correctly', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

});