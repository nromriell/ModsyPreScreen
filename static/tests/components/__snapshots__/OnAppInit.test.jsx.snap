// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnAppInit Tests Renders Correctly 1`] = `null`;
